GeoUtils/vector_tools.py

```diff
@@ -25,9 +25,15 @@ def __init__(self, filename):
         :return: A Vector object
         """
 
-        ds = gpd.read_file(filename)
-        self.ds = ds
-        self.name = filename
+        if isinstance(filename,str):
+            ds = gpd.read_file(filename)
+            self.ds = ds
+            self.name = filename
+        elif isinstance(filename,gpd.GeoDataFrame):
+            self.ds = filename
+            self.name = None
+        else:
+            raise ValueError('filename argument not recognised.')
```

Comment:
what kind of spelling is this?

Reply:
I think it's an old spelling from a country that once strove for independence !

Reply:
for my defense I copied/pasted this from raster_tools :p